### PR TITLE
Is this a rain?

### PR DESCRIPTION
**Pull request by Azrun**
## About the PR 
Added new dialog to Swampify:

image

- Yes.

rain
- No.
  - No change. 
- Particles
RainParticles
RainParticles2

This currently places an effect on each tile swampified turf that generates up to 8 particles from the upper left and will travel partway into south and east turf.
RainParticles3
 

Note: This currently causes reduced performance on the client when surrounded by that many particle generators. This seems to manifest in reduced responsiveness of other animations.  Glides are pretty chunky,

## Why's this needed? 
I want this for _Artemis_.  This is a great way for these features to be viewed and evaluated by admin/devs and for gimmickry.

### CODE REVIEW

**Comment by Azrun** [comment]
Hmmm, this also now has the particle effect I demoed for welding that is more of an example than anything.

File: code/obj/effects/particles.dm [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	alpha = 200
+
+	filters = list(type="bloom", threshold="#000", size=3, offset=2, alpha=1)
+
+	flame
+		filters = null
+		particles = new/particles/spark/flame
+
+	directed
+		particles = new/particles/spark/directed
```
Comment (Tarmunora) [inline code/obj/effects/particles.dm:63]: atomize PR

File: code/obj/effects/precipitation.dm [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	particles = new/particles/snow
+	plane = PLANE_NOSHADOW_ABOVE
+	client_attach
+		screen_loc = "CENTER"
+	dense
+		particles = new/particles/snow/dense
+	mega_dense
+		particles = new/particles/snow/mega_dense
+	grey
+		particles = new/particles/snow/grey
```
Comment (Tarmunora) [inline code/obj/effects/precipitation.dm:38]: Atomize PR

File: goonstation.dme [hunk 3]
```diff
@@ -1280,6 +1280,8 @@ var/datum/preMapLoad/preMapLoad = new
 #include "code\obj\effects\ion_trail.dm"
 #include "code\obj\effects\misc.dm"
 #include "code\obj\effects\mustard_gas.dm"
+#include "code\obj\effects\particles.dm"
```
Comment (Tarmunora) [inline goonstation.dme:1283]: Atomize PR

**Review by Tarmunora — CHANGES_REQUESTED** [review]

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**